### PR TITLE
Add mass operator to EagerDGDiscretization

### PR DESCRIPTION
**Pull request by majosm**
Adds a `mass()` operator method to `EagerDGDiscretization` (and fixes a couple of little bugs).

### CODE REVIEW

**Comment by inducer** [comment]
Is this draft still? (if so, please mark as such) I ask because:

- No tests
- Does not seem to add a mass operator

**Comment by majosm** [comment]
Debugging some test failures... might have fixed them now.

Testing via illinois-ceesd/mirgecom#180 at the moment. Not sure how to add tests for it here (I didn't see any for the other eager stuff that I could base off of).

File: grudge/symbolic/operators.py [hunk 1]
```diff
@@ -486,7 +486,7 @@ def __init__(self, dd_in=None, dd_out=None):
         if dd_in is None:
             dd_in = prim.DD_VOLUME
         if dd_out is None:
-            dd_out = dd_in
+            dd_out = prim.DD_VOLUME
```
Comment (majosm) [inline grudge/symbolic/operators.py:489]: @inducer This change broke a few things. I fixed the ones that were causing test failures, but I'm not positive that I got everything. Is there anything else I should check to be more confident that this is ok?

Comment (inducer) [reply grudge/symbolic/operators.py:489]: IMO: Tests are what provides confidence. If it wasn't tested, you may assume it was broken and don't need to worry about it.

File: grudge/eager.py [hunk 2]
```diff
@@ -83,6 +84,9 @@ def project(self, src, tgt, vec):
         :arg tgt: a :class:`~grudge.sym.DOFDesc`, or a value convertible to one
         :arg vec: a :class:`~meshmode.dof_array.DOFArray`
         """
+        if src == tgt:
```
Comment (inducer) [inline grudge/eager.py:87]:
```suggestion
        src = sym.as_dofdesc(src)
        tgt = sym.as_dofdesc(tgt)
        if src == tgt:
```

**Review by inducer — COMMENTED** [review]
LGTM. One minor suggestion, which I'll apply.

**Pull request merged**